### PR TITLE
Examples

### PR DESCRIPTION
**Pull request by kainino0x**

### CODE REVIEW

File: src/framework/fixture.ts [hunk 1]
```diff
@@ -43,11 +43,33 @@ export class Fixture {
...
   }
 
-  async shouldReject(p: Promise<unknown>, exceptionType?: string, msg?: string): Promise<void> {
+  protected async asyncExpectation(fn: () => Promise<void>): Promise<void> {
     this.numOutstandingAsyncExpectations++;
+    await fn();
+    this.numOutstandingAsyncExpectations--;
+  }
+
+  async shouldReject(p: Promise<unknown>, exceptionType?: string, msg?: string): Promise<void> {
```
Comment (austinEng) [inline src/framework/fixture.ts]: nit: If we don't care about the exceptionType, we'll have to call this as `shouldReject(p, undefined, msg)`
Thoughts on having the argument be a params dict `{ exceptionType?: string, msg?: string }` ?

File: src/suites/cts/examples.spec.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// - http://localhost:8080/?runnow=1&q=cts:examples:
+
+export const g = new TestGroup(GPUTest);
+
+g.test('basic', t => {
+  t.expect(true);
+  t.expect(true, 'true should be true');
+
+  t.shouldThrow(
+    () => {
```
Comment (austinEng) [inline src/suites/cts/examples.spec.ts]: unit: Will the formatter let this function be one line?

File: src/suites/cts/gpu_test.ts [hunk 3]
```diff
@@ -16,41 +16,43 @@ export class GPUTest extends Fixture {
...
-    }
+      if (size <= 256 && failedPixels > 0) {
+        const expHex = Array.from(exp)
+          .map(x => x.toString(16).padStart(2, '0'))
+          .join('');
+        const actHex = Array.from(actual)
+          .map(x => x.toString(16).padStart(2, '0'))
+          .join('');
+        this.rec.log('EXP: ' + expHex);
+        this.rec.log('ACT: ' + actHex);
```
Comment (austinEng) [inline src/suites/cts/gpu_test.ts]: nit: I'd prefer the full words EXPECT and ACTUAL.

File: src/framework/fixture.ts (hunk 1 above)
Comment (kainino0x) [reply src/framework/fixture.ts]: Decided to make the exception name required (and moved it to first position)

File: src/suites/cts/examples.spec.ts (hunk 2 above)
Comment (kainino0x) [reply src/suites/cts/examples.spec.ts]: Unfortunately not, it's very opinionated.

**Comment by kainino0x** [comment]
Thanks for the review! Merging now so I can send out the email, but feel free to send comments on the last few changes.

**Pull request merged**